core/state_processor.go

```diff
@@ -210,8 +210,7 @@ func handleMPS(ti int, tx *types.Transaction, gp *GasPool, usedGas *uint64, cfg
 func ApplyTransactionOnMPS(config *params.ChainConfig, bc *BlockChain, author *common.Address, originalGP *GasPool,
 	publicStateDBFactory func() *state.StateDB, privateStateDBFactory func(psi types.PrivateStateIdentifier) (*state.StateDB, error),
 	header *types.Header, tx *types.Transaction, usedGas *uint64, cfg vm.Config, privateStateRepo mps.PrivateStateRepository, applyOnPartiesOnly bool) (*types.Receipt, error) {
-	// clone the gas pool (as we don't want to keep consuming intrinsic gas multiple times for each MPS execution)
-	gp := new(GasPool).AddGas(originalGP.Gas())
```

Comment:
non-blocking: why before wasn’t needed a new gas pool for each PSI on MPS while now is needed? as PMT shouldn't impact this

Reply:
Whilst investigating an issue with the intrinsic gas change, I realised this was an issue on MPS. The cloned gaspool is needed for each PSI, otherwise the execution of the txn on each PSI will decrement the gaspool, potentially leading to `Error: gas limit reached` on the MPS node.

Reply:
ok. thanks. I need to check this also for the private pre fetching as I might have the same issue.